File: project/src/main/java/javaspark/sortbykeylianxi.java
//import org.apache.spark.SparkConf;
//import org.apache.spark.api.java.JavaPairRDD;
//import org.apache.spark.api.java.JavaSparkContext;
//import org.apache.spark.api.java.function.VoidFunction;
//import scala.Tuple2;
//import java.util.Arrays;
//import java.util.List;
//
//public class sortbykeylianxi {
//    public static void main(String[] args) {
//
//    }
//    private static void sortbykey(){
//        SparkConf conf=new SparkConf()
//                .setAppName("sortbykey")
//                .setMaster("local");
//        JavaSparkContext sc=new JavaSparkContext(conf);
//        List<Tuple2<Integer,String>> sortList= Arrays.asList(
//                new Tuple2<Integer,String>(12,"leo"),
//                new Tuple2<Integer,String>(23,"tom"),
//                new Tuple2<Integer,String>(43,"le"),
//                new Tuple2<Integer,String>(54,"lo"),
//                new Tuple2<Integer,String>(45,"javk"),
//                new Tuple2<Integer,String>(3,"marry"),
//                new Tuple2<Integer,String>(43,"jeson"));
//        JavaPairRDD<Integer,String> scoresort=sc.parallelizePairs(sortList);
//        JavaPairRDD<Integer,String> sortScore=scoresort.sortByKey();
//        sortScore.foreach(new VoidFunction <Tuple2 <Integer, String>>() {
//            @Override
//            public void call(Tuple2 <Integer, String> t) throws Exception {
//                System.out.println(t._1+":"+t._2);
//            }
//        });
//        sc.close();
//    }
//}

File: project/src/main/scala/day20190319/sortbykeylianxi.scala
package day20190319
import org.apache.spark.{SparkConf, SparkContext}

object sortbykeylianxi {
  def main(args: Array[String]): Unit = {
    sortbylian()
  }
  def sortbylian(): Unit ={
    val conf=new SparkConf()
      .setAppName("sortbykey")
      .setMaster("local")
    val sc=new SparkContext(conf)
    val scoreList=Array(Tuple2(43,"leo"),Tuple2(44,"tom"),Tuple2(12,"marry"),
                 Tuple2(42,"jeason"),Tuple2(56,"jim"))
    val scores=sc.parallelize(scoreList,1)
    val sortedScores = scores.sortByKey(false)
    sortedScores.foreach(sortedScores=>println(sortedScores._1+":"+sortedScores._2))
  }
}
